### DISCUSSION TITLE
Loki: Improve error handling when user submits a blank query (whitespace)

### DISCUSSION THREAD

**Discussion by SanthoshS20** [grafana/grafana#53999] [Observability]
**What happened**:
In the Loki Query Dashboard, Query field is not properly validated. If its only have one or more spaces in a query, it executes and show parse error. I tried by executing the query with only having a space in the query field and results in parse error.

**What you expected to happen**:
It should validate whether the query field have a query or not. If its doesn't have a query, it should show the warning message **Query field is empty** and should not execute any API calls.

**How to reproduce it (as minimally and precisely as possible)**:
**Step 1:** Go to the Loki Query Dashboard.
**Step 2:** In the first query field, add a space.
**Step 3:** Run the query.

**Anything else we need to know?**:
![image](https://user-images.githubusercontent.com/37366893/185418672-a8551f49-ebb5-4148-a21f-cbd5e89b6648.png)

**Environment**:
- Grafana version: 9.0.7
- User OS & Browser: Windows 10 & Google Chrome

**Comment by gabor** [comment]
hi @SanthoshS20 thanks for reporting this. yes, currently there is no validation of the query on grafana's side. (feature requests are handled in the discussions-section, i'll move this issue there too.)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
